### PR TITLE
Pattern recog engine

### PR DESCRIPTION
**Pull request by ErnieAtLYD**
## Summary by Sourcery

Transform the plugin into a modular, service-based architecture and introduce end-to-end journal analysis features

New Features:
- Add AES-256 encryption service with master-password setup and management UI
- Implement CacheService with TTL, disk persistence, and cleanup for analysis caching
- Introduce PatternRecognitionService to detect behavioral patterns and trends
- Add AnalysisManager to orchestrate pattern, trend, and comprehensive AI-powered journal analysis
- Expose new commands for pattern analysis, trend analysis, comprehensive analysis, and cache clearing
- Extend FileOperationsService to generate and save formatted analysis report files

Enhancements:
- Refactor main plugin into ServiceManager-driven dependency injection for modularity and maintainability
- Update AIService to support custom prompt response generation
- Allow API key storage as plain text or encrypted data with seamless decryption handling
- Restructure code into separate services, modals, and indexed exports for clarity

Build:
- Add test, test:watch, and test:coverage scripts to package.json
- Include Babel configuration for JavaScript transformation

CI:
- Configure Jest for unit testing with ESM and Babel support
- Add comprehensive unit tests and Obsidian API mocks for EncryptionService

Documentation:
- Revise README and CLAUDE.md to document new encryption, caching, pattern/trend analysis, and service-based architecture

Tests:
- Provide extensive EncryptionService suite covering key derivation, encryption/decryption cycles, password validation, and edge cases

### CODE REVIEW

File: src/services/CacheService.ts [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+                oldestKey = key;
+            }
+        }
+
+        if (oldestKey) {
+            this.cache.delete(oldestKey);
+        }
+    }
+
+    private async loadFromDisk(): Promise<void> {
```
[the remark this answers is not in the document]
Comment (ErnieAtLYD) [reply src/services/CacheService.ts:260]: @claude Implement this, please.
